node/network/availability-recovery/src/lib.rs

```diff
@@ -304,7 +304,20 @@ impl RequestChunksFromValidators {
 		sender: &mut impl SubsystemSender,
 	) {
 		let num_requests = self.get_desired_request_count(params.threshold);
-		let mut requests = Vec::with_capacity(num_requests - self.requesting_chunks.len());
+		let candidate_hash = &params.candidate_hash;
+		let already_requesting_count = self.requesting_chunks.len();
+
+		gum::debug!(
+			target: LOG_TARGET,
+			?candidate_hash,
+			?num_requests,
+			error_count= ?self.error_count,
+			total_received = ?self.total_received_responses,
+			threshold = ?params.threshold,
+			?already_requesting_count,
+			"Requesting availability chunks for a candidate",
+		);
+		let mut requests = Vec::with_capacity(num_requests - already_requesting_count);
```

Comment:
nit: with complexity of `get_desired_request_count` I'd prefer a `saturating_sub` here